saleor/graphql/discount/mutations/voucher/voucher_delete.py

```diff
@@ -38,6 +38,24 @@ def success_response(cls, instance):
         return response
 
     @classmethod
-    def post_save_action(cls, info: ResolveInfo, instance, cleaned_input):
+    def post_save_action(cls, info: ResolveInfo, instance, code):
         manager = get_plugin_manager_promise(info.context).get()
-        cls.call_event(manager.voucher_deleted, instance)
+        cls.call_event(manager.voucher_deleted, instance, code)
+
+    @classmethod
+    def perform_mutation(  # type: ignore[override]
+        cls, _root, info: ResolveInfo, /, *, external_reference=None, id=None
+    ):
+        """Perform a mutation that deletes a model instance."""
+        instance = cls.get_instance(info, external_reference=external_reference, id=id)
+
+        cls.clean_instance(info, instance)
+        db_id = instance.id
+        code = instance.codes.first().code
```

Comment:
What if the voucher will have multiple codes?

Reply:
I understand that you're passing the code to match the static webhook payload, but what about subscription `VoucherDeleted`? The codes don't exist as you deleted the `Voucher`, so I will not be able to fetch the codes that were removed.  Is it something that we should take into account?
Also, I think we should double-check (if not yet checked) the subscription query for `VoucherDeleted` and requested `codes` field

Reply:
A talk with @IKarbowiak and we will add events related to voucher codes as separate PR(https://github.com/saleor/saleor/issues/14495) after we merge this. I don't think there is an easy way to reassign deleted related objects to voucher. So we think a dedicated webhook is the best option. What do you think @korycins ?

Reply:
I just wanted to confirm that we have taken into account this. If you have addressed this then it is ok for me.